wallet/src/de/schildbach/wallet/ui/InputParser.java

```diff
@@ -173,11 +167,34 @@ public void parse() {
 
                     error(x, R.string.input_parser_invalid_transaction, x.getMessage());
                 }
-            } else {
+            } else if (!tryFindAnyMatch(input)) {
                 cannotClassify(input);
             }
         }
 
+        private boolean tryFindAnyMatch(String input) {
+            Matcher matcher = PATTERN_BITCOIN_ADDRESS.matcher(input);
+
+            if (matcher.find() && matcher.group(0) != null) {
+                try {
+                    String addressStr = matcher.group(0);
+                    assert addressStr != null;
+                    final Address address = Address.fromString(Constants.NETWORK_PARAMETERS, addressStr);
+                    PaymentIntent intent = PaymentIntent.fromAddress(address, null);
+                    intent.setShouldConfirmAddress(true);
+
+                    handlePaymentIntent(intent);
+                } catch (final AddressFormatException x) {
+                    log.info("got invalid address", x);
+                    error(x, R.string.input_parser_invalid_address);
+                }
+
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
Yes, it's my thinking too - request URL is less likely to get tangled together with other text.

Pattern matching `dash:<address>&amount=<amount>` would be more complicated indeed (and I'm no expert in regex). Additionally, it would not work for the case when the request itself has an error in it (like with that bad symbol Andrei had in his testing).

I think extracting a correct dash address should be enough here.